### PR TITLE
Changes to support RAID

### PR DESCRIPTION
**Pull request by rexcsn**
If merged, the following changes will take place:
- Changed cfnconfig.py to parse all RAID related variables
- Changed config_sanity.py to check RAID related constraints
- Changed aws-parallelcluster.cfn.json to add new parameter RAIDoptions, with allowed pattern to validate input, and new substack RAIDSubstack, output RAID shared directory via dna.json to master and compute nodes, output RAID volume ids to master. Added new parameter RAIDSharedDir to Batch substack resource
- Changed batch-substack.cfn.json to set RAIDSharedDir as an environment variable so the RAID directory can be mounted
- Added util/generate-raid-substack.py, which generates raid substack template with troposphere
- Added raid-substack.cfn.json, generated with generate-raid-substack.py, supports creating up to 5 volumes
- Changed entrypoint.sh to include RAID directory mounting call
- Added integration test raid-test.py and test files config_raid and raid-check.sh, which tests edge cases of creating with the minimum number of 2 volumes and max number of 5 volumes, as well as the default case of not using RAID at all.
- Batch compatibility is manually tested by submitting a job under RAID shared directory

Signed-off-by: Rex Chen <shuningc@amazon.com>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: util/generate-raid-substack.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        if i == 0:
+            outputs[i] = If(use_vol[i], vol_to_return[i], "NONE")
+        else:
+            outputs[i] = If(use_vol[i], Join(",", vol_to_return[: (i + 1)]), outputs[i - 1])
+
+    t.add_output(
+        Output("Volumeids", Description="Volume IDs of the resulted RAID EBS volumes", Value=outputs[number_of_vol - 1])
+    )
+
+    json_file_path = "/Users/shuningc/Desktop/pcluster-github/cfncluster/cloudformation/raid-substack.cfn.json"
```
Comment (fnubalaj) [inline util/generate-raid-substack.py]: This should be an input. Don't hardcode it.

File: cli/pcluster/resources/batch/docker/scripts/entrypoint.sh [hunk 2]
```diff
@@ -18,6 +18,11 @@ if [[ ! -z "${EFS_FS_ID}" ]] && [[ ! -z "${AWS_REGION}" ]] && [[ "${EFS_SHARED_D
   /parallelcluster/bin/mount_efs.sh "${EFS_FS_ID}" "${AWS_REGION}" "${EFS_SHARED_DIR}"
 fi
 
+# mount RAID via nfs
+if [ ! -z ${RAID_SHARED_DIR} ]; then
```
Comment (fnubalaj) [inline cli/pcluster/resources/batch/docker/scripts/entrypoint.sh]: If there is always going to be a RAID_SHARED_DIR? If the user does not request RAID, will there be RAID_SHARED_DIR?
